spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestEngine.kt

```diff
@@ -31,10 +35,34 @@ class SpekTestEngine : TestEngine {
             .map { Paths.get(it) }
             .map { it.toString() }
 
-        val pathSelector = discoveryRequest.getSelectorsByType(SpekPathDiscoverySelector::class.java)
-            .firstOrNull() ?: SpekPathDiscoverySelector(PathBuilder.ROOT)
+        val classSelectors = discoveryRequest.getSelectorsByType(ClassSelector::class.java)
+            .map {
+                val packageName = it.javaClass.`package`.name
+                val className = it.javaClass.name.removePrefix("$packageName.")
 
-        val discoveryResult = runtime.discover(DiscoveryRequest(sourceDirs, listOf(pathSelector.path)))
+                PathBuilder()
+                    .append(packageName)
+                    .append(className)
+                    .build()
+            }
+
+        val packageSelectors = discoveryRequest.getSelectorsByType(PackageSelector::class.java)
+            .map {
+                PathBuilder().append(it.packageName)
+                    .build()
+            }
+
+        val filters = mutableListOf<Path>()
+
+        filters.addAll(classSelectors)
+        filters.addAll(packageSelectors)
+
+        // todo: empty filter should imply root
```

Comment:
To minimize the scope of this change, will be addressing it in another PR. Basically just don't filter if list is empty.